### DISCUSSION TITLE
Unable to change the type of name PUSH! from VALUE to METHOD.

### DISCUSSION THREAD

**Discussion by Inc0n** [coalton-lang/coalton#772] [Help!] [answered]
Admist my development on my little project using coalton. I am having problem with the following:
```
Unable to change the type of name PUSH! from VALUE to METHOD.
```
So I had `push` defined as a normal function prior, and now wish it to be part of a `define-class`.
How can I make this happen without restarting my slime repl?

**Comment by eliaslfox** [comment, marked as answer]
It's not currently possible. Changing a method to a value would require un-defining the method's class and all of that classes instances. This is one of several places where new definitions can significantly conflict Coalton's internal state. The eventual goal is to resolve these with restarts. See #655 for another example.

**Reply by Inc0n** [reply]
I see, an inherent problem to typed languages.  Thanks for answering!
